src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -1201,11 +1201,12 @@ bool IdealLoopTree::policy_range_check(PhaseIdealLoop* phase, bool provisional,
 
       // Comparing trip+off vs limit
       Node *bol = iff->in(1);
-      if (bol->req() != 2) {
+      if (bol->req() < 2) {
         continue; // dead constant test
```

Comment:
Before: We bailed out for `Opaque4` nodes because they have 3 inputs. But comment suggests that this is only for dead constant tests. I therefore updated this and added assert below for `Opaque4` nodes.